src/error.rs

```diff
@@ -60,6 +60,15 @@ pub(crate) fn unsigned_variant_to_multihash_error(err: unsigned_varint::io::Read
     }
 }
 
+#[cfg(not(feature = "std"))]
+impl From<unsigned_varint::decode::Error> for Error {
```

Comment:
@thomaseizinger I've implemented this with the `From` trait. Are there any reasons doing it with a function like the other conversions?

I'm even thinking about doing a PR switching the other ones to from traits as well, which seems more idiomatic to me.

Reply:
The `From` trait implementation makes this part of the public API. Given that `unsigned_varint` is not yet `1.0`, I'd advise against it because it would mean that we have to also issue a new minor version if we update `unsigned_varint` to a new minor version.

Reply:
It is more idiomatic but unfortunately a semver hazard because trait implementations don't stop at the crate boundary.

Reply:
> Though I don't really like it.

I can relate. I also don't like the verbosity. Although I think the trade-off for being able update dependencies without cascading breaking changes up is definitely worth it.